src/adapters/device/DeviceLocale.ts

```diff
@@ -0,0 +1,16 @@
+// import { NextPageContext } from 'next'
+import { IDeviceLocale } from '../../core/useCases/locale'
+import { Locale } from '../../core/useCases/locale/locale.reducer'
+
+import { isSSR } from '../../utils/misc'
+
+export class DeviceLocale implements IDeviceLocale {
+  getDeviceLocale(): Locale {
+    let browserLocale = null
+    if (!isSSR) {
+      browserLocale = Array.isArray(navigator.languages) ? navigator.languages[0] : navigator.languages
+    }
+
+    return browserLocale ? browserLocale.substr(0, 2) : 'fr'
+  }
```

Comment:
Ça ne serait pas plus simple d'utiliser `navigator.language` ?
D'ailleurs on n'est pas censé gérer de valeur par défaut ici, on doit juste retourner la locale du navigateur. La valeur par défaut est géré dans notre code métier.
Je pense que ça peut suffire ça, non ?

```suggestion
    return !isSSR
      ? navigator.language.substr(0, 2)
      : null
    }
```